pkg/helm/merge.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+	"reflect"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// merge will do a simple merge of an array of yaml documents. If there are conflicts,
+// yaml documents that appear later in the array will override those that appear earlier.
+// Sequence nodes will concatenated, while map nodes will be merged together.
+// Example:
+// The following two yaml documents:
+//
+//	```
+//	namespaces:
+//	  - name: foo
+//	    rolebindings:
+//	      name: foo-rb
+//	```
+//	```
+//	namespaces:
+//	  - name: bar
+//	    rolebindings:
+//	      name: bar-rb
+//	```
+//
+// Will get merged into the following:
+//
+//	```
+//	namespaces:
+//	  - name: foo
+//	    rolebindings:
+//	      name: foo-rb
+//	  - name: bar
+//	    rolebindings:
+//	      name: bar-rb
+//	```
+func merge(valuesToMerge [][]byte) ([]byte, error) {
```

Comment:
Can you add unit tests for each of these merge functions?

Reply:
done